src/cli/cmd/query/dependencies/dependencies.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) CloudCover
+// SPDX-License-Identifier: Apache-2.0
+
+package dependencies
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/MakeNowJust/heredoc/v2"
+	"github.com/cldcvr/terrarium/src/cli/internal/config"
+	"github.com/cldcvr/terrarium/src/pkg/db"
+	"github.com/cldcvr/terrarium/src/pkg/pb/terrariumpb"
+	"github.com/cldcvr/terrarium/src/pkg/transporthelper"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+)
+
+var (
+	cmd *cobra.Command
+
+	flagOutputFormat string
+	flagPageSize     int32
+	flagPageIndex    int32
+	flagSearchText   string
+)
+
+func NewCmd() *cobra.Command {
+	cmd = &cobra.Command{
+		Use:   "dependencies",
+		Short: "List dependency details matching the dependency interface id",
+		Long: heredoc.Docf(`
+		The 'dependencies' command provides detailed information about the available dependencies.
+		When invoked without any filters, it lists all the dependencies present. You can also refine
+		the list using optional search criteria.
+
+		Usage examples:
+	  		terrarium dependencies                             // Lists all dependencies
+	  		terrarium dependencies -s <SEARCH_TEXT>  // Filters the list based on the search text
+
+		The above commands will fetch and display the details of the dependencies matching the criteria,
+		if provided.
+		`),
+		RunE: fetchDependencies,
+	}
+
+	cmd.Flags().StringVarP(&flagSearchText, "searchText", "s", "", "optional search text")
+	cmd.Flags().Int32Var(&flagPageSize, "pageSize", 100, "page size flag")
+	cmd.Flags().Int32Var(&flagPageIndex, "pageIndex", 0, "page index flag")
+	cmd.Flags().StringVarP(&flagOutputFormat, "output", "o", "json", "Output format (json or table)")
+
+	return cmd
+}
+
+func fetchDependencies(cmd *cobra.Command, args []string) error {
+	g, err := config.DBConnect()
+	if err != nil {
+		return err
+	}
+
+	page := &terrariumpb.Page{
+		Size:  flagPageSize,
+		Index: flagPageIndex,
+	}
+
+	dbDependencies, err := g.FetchAllDependency(
+		db.DependencySearchFilter(flagSearchText),
+		db.PaginateGlobalFilter(page.Size, page.Index, &page.Total),
+	)
+	if err != nil {
+		return err
+	}
+
+	// Convert each db.Dependency in dbDependencies to its protobuf representation
+	var result []*terrariumpb.Dependency
+	for _, dbDep := range dbDependencies {
+		protoDep, err := dbDep.ToProto()
+		if err != nil {
+			return err
+		}
+		result = append(result, protoDep)
+	}
```

Comment:
It'd be more useful to have a `ToProto` function on the `Dependencies` array type instead of running the loop here. Like we have on other db modals: https://github.com/cldcvr/terrarium/blob/main/src/pkg/db/module.go#L159
This helps in maintaining the consistency, and in future if there anything we need to change in proto translation, or we want to make that translation somewhere else like on API, we can expect to find that logic in db layer, ToProto func.